tests/ttnn/unit_tests/operations/test_prepare_conv_weights.py

```diff
@@ -0,0 +1,177 @@
+    conv_weight_shape = (output_channels, input_channels, filter_height, filter_width)
+    torch_weight_tensor = torch.randn(conv_weight_shape, dtype=torch.bfloat16)
+    torch_input_tensor = torch.randn(inp_shape, dtype=torch.bfloat16)
+    torch_bias_tensor = torch.randn((1, 1, 1, output_channels), dtype=torch.bfloat16) if has_bias else None
+
+    torch_out_golden_tensor = torch.nn.functional.conv2d(
+        torch_input_tensor,
+        torch_weight_tensor,
+        bias=torch_bias_tensor.reshape(-1) if has_bias else None,
+        stride=(stride_h, stride_w),
+        padding=(pad_h, pad_w),
+        dilation=(1, 1),
+        groups=1,
+    ).permute(0, 2, 3, 1)
+
+    tt_input_tensor = ttnn.from_torch(torch_input_tensor.transpose(-3, -2).transpose(-2, -1), ttnn.bfloat16)
+    tt_weight_tensor = ttnn.from_torch(torch_weight_tensor, ttnn.bfloat16)
+    tt_bias_tensor = ttnn.from_torch(torch_bias_tensor, ttnn.bfloat16) if has_bias else None
+
+    conv_config = ttnn.Conv2dConfig(
+        dtype=ttnn.bfloat16,
+        weights_dtype=ttnn.bfloat16,
+        input_channels_alignment=(16 if input_channels == 16 and input_height == 115 else 32),
+        packer_l1_accum_enabled=packer_l1_acc,
+        enable_act_double_buffer=False,
+        enable_split_reader=False,
+        enable_subblock_padding=False,
+    )
+
+    if config_override and "act_block_h" in config_override:
+        conv_config.act_block_h_override = config_override["act_block_h"]
+
+    if config_override and "act_block_w_div" in config_override:
+        conv_config.act_block_w_div = config_override["act_block_w_div"]
+
+    if config_override and "num_cores_nhw" in config_override:
+        if config_override["num_cores_nhw"] == 98:
+            conv_config.core_grid = ttnn.CoreRangeSet({ttnn.CoreRange((0, 0), (11, 7)), ttnn.CoreRange((0, 8), (1, 8))})
+            conv_config.override_sharding_config = True
+            print("Setting num_cores_nhw to 98")
+
+    conv_kwargs = {
+        "in_channels": input_channels,
+        "out_channels": output_channels,
+        "batch_size": batch_size,
+        "input_height": input_height,
+        "input_width": input_width,
+        "kernel_size": (filter_height, filter_width),
+        "stride": (stride_h, stride_w),
+        "padding": (pad_h, pad_w),
+        "dilation": (1, 1),
+        "groups": 1,
+        "device": device,
+        "conv_config": conv_config,
+    }
+
+    tt_weight_tensor_formatted = ttnn.prepare_conv_weights_for_ttnn(
+        weight_tensor=tt_weight_tensor, weights_format="OIHW", **conv_kwargs
+    )
+    tt_bias_tensor_formatted = (
+        ttnn.prepare_conv_bias_for_ttnn(bias_tensor=tt_bias_tensor, **conv_kwargs) if has_bias else None
+    )
+
+    tt_weight_tensor_formatted = ttnn.to_device(tt_weight_tensor_formatted, device)
+    tt_bias_tensor_formatted = ttnn.to_device(tt_bias_tensor_formatted, device) if has_bias else None
+
+    tt_output_tensor_on_device = ttnn.conv2d_device_weights(
```

Comment:
I don't think we want to have conv2d_device_weights and conv2d_host_weights
our conv2d should just handle inputs it gets the right way, at least that is my understanding
@mywoodstock @ayerofieiev-tt  what is your take here

Reply:
The reason I made it this way is because the old `conv2d` returns a tuple (that has the output, formatted weight, formatted bias). And this new `conv2d` will not. Personally I think that the old conv2d should be eliminated in another PR. All tests involving conv2d though will have to add the weight/bias preparation ops so I think it would be best to do that in another PR since the diff would span many files.

Reply:
Yeah, thats the plan -- eventually replace the conv2d with this version -- As long as we use the same implementation for formatting the weights, this will give an option to format it first. The flow in conv2d.cpp will remain the same since we have a check to not re-format weights if they are already on device.

Reply:
Is it two bad if new version returns a tupple for now even if doesn't make too much sense?
Do we have a plan on who is going to do the clean up? @LPanosTT do you plan to do this as a follow up?